src/main/java/Ui.java

```diff
@@ -0,0 +1,40 @@
+import java.io.IOException;
+import java.util.Scanner;
+
+public class Ui {
+
+    public Ui() {
+    }
+
+    public void hello() {
+        String openingMessage = "\tHello! I'm Duke\n\tWhat can I do for you?\n";
+        System.out.println(openingMessage);
+    }
+
+    public void takeInUserInput() {
+        Scanner scanner = new Scanner(System.in);
+        Parser parser = new Parser();
+        while (scanner.hasNext()) {
+            String command = scanner.nextLine();
+            command = command.trim();
+
+            try {
+                parser.executeCommand(command);
+            } catch (InvalidCommandException | MissingInputException | MissingDescriptionException e) {
```

Comment:
Great use of inheritance for exception handling! Makes your code look very neat.